### ISSUE TITLE
Custom indicators for URL and Actions

### ISSUE THREAD

**Issue by Sweenu** [dunst-project/dunst#509]
I'd love to have the opportunity to change my notifications' colors or frame if they have actions or URLs.
Or at least, be able to change `(A)` and `(U)` to something else.

**Comment by ismig** [comment]
I find the (AU) fairly distracting. It's so large it dwarfs the appname of my notifications, which is deliberately small and grey.
Small icons would be less distracting and easier to interpret -- maybe the link icon and something else for actions (a hammer? lol). Or just a custom format string.

**Comment by Kryszak** [comment]
Simply being able to set indicators to some other string stylable with HTML would be great!

**Comment by tsipinakis** [comment]
I have a faint memory of discussing this with @bebehei but I can't seem to find any logs of it.

How about adding conditionals to the `format` setting for this?
For example the default format becomes something like `%?action?(A)? %?url?(U)? %?both?(AU)?`

Though I don't like that we have to use `both` as well to keep the behaviour consistent. This also raises the question of how to add this without breaking indicators for all existing configs.

**Comment by bebehei** [comment]
@tsipinakis Yeah, we had been discussing previously about this. But I'm not sure, where. I know I had been starting to implement similar stuff with a date field (IIRC I closed this PR because I didn't find an acceptable solution).

**Comment by monkz** [comment]
For my part, i would like to customize them, to add the count of actions that is bound to a notification - Like A3. And for U replace it with a emoji.

**Comment by elamperti** [comment]
Hi everyone, I've drafted [a PR](https://github.com/dunst-project/dunst/pull/1150) with changes to have custom indicators. I tried it with emoji and it works (and looks quite nice!). The indicators can be styled in the format string using Pango markup.

I'd love to hear your feedback about the implementation.

---

@tspinakis from what I saw in the code, having conditionals in the format string would be probably difficult to implement (and/or would end up requiring an ugly syntax), so I'd discard that for now.

@monkz this implementation doesn't show the number of actions or URLs, it could be added but I don't know if it would be generally useful. What would be a use case that favors knowing the number of actions at first glance?

**Comment by monkz** [comment]
@elamperti This allows me to determine if I need the context menu or just a click on the notification.
